### PR TITLE
CONCF-673 Send video object with Main page also

### PR DESCRIPTION
**Pull request by hakubo**
https://wikia-inc.atlassian.net/browse/CONCF-673

### CODE REVIEW

**Comment by hakubo** [comment]
@bognix @rogatty @Warkot @vforge 
This is my initial idea
please focus on high level Code Review

I think createMediaObject should be moved to better place, where? Or maybe it is a good place since this function does create an object that represents a video object to be included in any json response. Or should I create Media class or something similar?

File: extensions/wikia/ArticleAsJson/ArticleAsJson.class.php [hunk 1]
```diff
@@ -40,22 +39,17 @@ private static function createMediaObj( $details, $imageName, $caption = '', $li
...
-
-		if ( !empty( $details['width'] ) ) {
-			$media['width'] = (int) $details['width'];
+		if ( is_string( $caption ) && $caption !== '' ) {
+			$media['caption'] = $caption;
 		}
 
-		if ( !empty( $details['height'] ) ) {
-			$media['height'] = (int) $details['height'];
+		if ( is_string( $link ) && $link !== '' ) {
```
Comment (bognix) [inline extensions/wikia/ArticleAsJson/ArticleAsJson.class.php]: why not using !empty($link) ?

File: extensions/wikia/ArticleAsJson/ArticleAsJson.class.php [hunk 2]
```diff
@@ -40,22 +39,17 @@ private static function createMediaObj( $details, $imageName, $caption = '', $li
 			$media['context'] = $context;
 		}
 
-		if ( is_string( $link ) && $link !== '' ) {
-			$media['link'] = $link;
-		}
-
-		if ( !empty( $details['width'] ) ) {
-			$media['width'] = (int) $details['width'];
```
Comment (rogatty) [inline extensions/wikia/ArticleAsJson/ArticleAsJson.class.php]: Why did you remove it? Won't it break anything?

File: extensions/wikia/ArticleAsJson/ArticleAsJson.class.php [hunk 3]
```diff
@@ -40,22 +39,17 @@ private static function createMediaObj( $details, $imageName, $caption = '', $li
...
-		if ( !empty( $details['height'] ) ) {
-			$media['height'] = (int) $details['height'];
+		if ( is_string( $link ) && $link !== '' ) {
+			$media['link'] = $link;
 		}
 
 		if ( $details['mediaType'] == 'video' ) {
 			$media['views'] = (int) $details['videoViews'];
 			$media['embed'] = $details['videoEmbedCode'];
-			$media['provider'] = $details['providerName'];
```
Comment (rogatty) [inline extensions/wikia/ArticleAsJson/ArticleAsJson.class.php]: Why?

**Comment by rogatty** [comment]
@hakubo I don't have any better idea where to put `createMediaObject` method, sorry.

This sounds reasonable:
>this function does create an object that represents a video object to be included in any json response

File: extensions/wikia/ArticleAsJson/ArticleAsJson.class.php (hunk 2 above)
Comment (hakubo) [inline extensions/wikia/ArticleAsJson/ArticleAsJson.class.php]: it is send in embed object already

File: extensions/wikia/ArticleAsJson/ArticleAsJson.class.php (hunk 3 above)
Comment (hakubo) [inline extensions/wikia/ArticleAsJson/ArticleAsJson.class.php]: it is send in embed object already

**Comment by rogatty** [comment]
LGTM

File: extensions/wikia/ArticleAsJson/ArticleAsJson.class.php (hunk 1 above)
Comment (Warkot) [inline extensions/wikia/ArticleAsJson/ArticleAsJson.class.php]: What if the link equals `0` like on http://fallout.wikia.com/wiki/0?

Comment (hakubo) [inline extensions/wikia/ArticleAsJson/ArticleAsJson.class.php]: and that would be a good reason! :) Thanks

Comment (Warkot) [inline extensions/wikia/ArticleAsJson/ArticleAsJson.class.php]: De nada! :sunglasses:

File: extensions/wikia/MercuryApi/models/MercuryApi.class.php [hunk 4]
```diff
@@ -370,6 +370,30 @@ public function processTrendingData( $data, $itemArrayName, $paramsToInclude = [
...
+		$videosData = $data[ 'videos' ];
+
+		if ( !isset( $videosData ) || !is_array( $videosData ) ) {
+			return null;
+		}
+
+		$items = [];
+
+		foreach ( $videosData as $item ) {
+			$items[] = ArticleAsJson::createMediaObject(
```
Comment (hakubo) [inline extensions/wikia/MercuryApi/models/MercuryApi.class.php]: Looks like this function return embed code the same for all videos...

**Comment by vforge** [comment]
assuming CONCF-674 fix, LGTM

**Comment by bognix** [comment]
ping @hakubo @rogatty @vforge

**Comment by bognix** [comment]
Story created for implementing removal of width,height and providerName properly: https://wikia-inc.atlassian.net/browse/CONCF-703

**Comment by vforge** [comment]
ha! :+1: LGTM

**Comment by tomasznapieralski** [comment]
LGTM
